include/emotion.hpp

```diff
@@ -0,0 +1,47 @@
+#include <math.h>
+#include <functional>
+#include <vector>
+#include <map>
+#include <array>
+#include <string>
+#include <iostream>
+
+#include <Arduino.h>
+#include <ArduinoJson.h>
+#include <SPIFFS.h>
+
+#include "eye_assets.hpp"
+
+#if defined(STAMPS3)
+#include <lgfx_round_lcd_stamp_s3.hpp>
+#elif defined(STAMPC3)
+#include <lgfx_round_lcd_stamp_c3.hpp>
+#endif
+
+class EmotionManager
+{
+private:
+    EyeManager& eyeManager; // EyeManager の参照を保持
+    int current_eye_status = -1;
+    int frame = 0;
+
+public:
+    // コンストラクタで EyeManager の参照を受け取る
+    EmotionManager(EyeManager& manager) : eyeManager(manager) {}
+
+    // 感情ごとの表示設定
+    void set_emotion(int eye_status) {
+      if (current_eye_status != eye_status && eye_assets_map.find(eye_status) != eye_assets_map.end()) {
```

Comment:
Basically, you shouldn't refer to global variables inside of classmethod to capsule relevant data into the class.